test/e2e/metrics_test.go

```diff
@@ -134,79 +135,90 @@ var _ = Describe("Metrics", func() {
 	})
 
 	Context("When reaching the pipeline limit", Ordered, func() {
-		allPipelines := make(map[string][]client.Object, 0)
+		pipelinesObjects := make(map[string][]client.Object, 0)
+		pipelines := kyma.NewPipelineList()
 
 		BeforeAll(func() {
 			for i := 0; i < maxNumberOfMetricPipelines; i++ {
-				pipelineName := fmt.Sprintf("pipeline-%d", i)
-				pipelineObjects := makeBrokenMetricPipeline(pipelineName)
-				allPipelines[pipelineName] = pipelineObjects
+				objs, name := makeBrokenMetricPipeline(fmt.Sprintf("pipeline-%d", i))
+				pipelines.Append(name)
+				pipelinesObjects[name] = objs
 
-				Expect(kitk8s.CreateObjects(ctx, k8sClient, pipelineObjects...)).Should(Succeed())
+				Expect(kitk8s.CreateObjects(ctx, k8sClient, objs...)).Should(Succeed())
 			}
 
 			DeferCleanup(func() {
-				for _, pipeline := range allPipelines {
-					Expect(kitk8s.DeleteObjects(ctx, k8sClient, pipeline...)).Should(Succeed())
+				for _, objs := range pipelinesObjects {
+					Expect(kitk8s.DeleteObjects(ctx, k8sClient, objs...)).Should(Succeed())
 				}
 			})
 		})
 
 		It("Should have only running pipelines", func() {
-			for pipelineName := range allPipelines {
-				metricPipelineShouldBeRunning(pipelineName)
-				metricPipelineShouldBeDeployed(pipelineName)
+			idx := 0
+			for range pipelinesObjects {
+				metricPipelineShouldBeRunning(pipelines.At(idx))
+				metricPipelineShouldBeDeployed(pipelines.At(idx))
 			}
 		})
 
 		It("Should have a pending pipeline", func() {
 			By("Creating an additional pipeline", func() {
-				newPipelineName := "new-pipeline"
-				newPipeline := makeBrokenMetricPipeline(newPipelineName)
-				allPipelines[newPipelineName] = newPipeline
+				newObjs, newName := makeBrokenMetricPipeline("exceeding-pipeline")
+				pipelinesObjects[newName] = newObjs
+				pipelines.Append(newName)
 
-				Expect(kitk8s.CreateObjects(ctx, k8sClient, newPipeline...)).Should(Succeed())
-				metricPipelineShouldStayPending(newPipelineName)
-				metricPipelineShouldNotBeDeployed(newPipelineName)
+				Expect(kitk8s.CreateObjects(ctx, k8sClient, newObjs...)).Should(Succeed())
+				metricPipelineShouldStayPending(newName)
+				metricPipelineShouldNotBeDeployed(newName)
 			})
 		})
 
 		It("Should have only running pipeline", func() {
 			By("Deleting a pipeline", func() {
-				deletedPipeline := allPipelines["pipeline-0"]
-				delete(allPipelines, "pipeline-0")
+				deletedPipeline := pipelinesObjects[pipelines.First()]
+				delete(pipelinesObjects, pipelines.First())
 
 				Expect(kitk8s.DeleteObjects(ctx, k8sClient, deletedPipeline...)).Should(Succeed())
-				for pipelineName := range allPipelines {
-					metricPipelineShouldBeRunning(pipelineName)
+				idx := 0
```

Comment:
Why do you start with 0 as you are skipping 0 again in the `for` loop later.

Reply:
That is because I iterate over `pipelinesObjects` map to get access to the `pipelines`'s array elements. I though that index manipulations will make code more cryptic.

Reply:
Reimplemented the iterator